example/src/App.tsx

```diff
@@ -6,28 +6,30 @@ import {
   StyleSheet,
   View,
 } from 'react-native';
-import TrackPlayer from 'react-native-track-player';
+import TrackPlayer, { useActiveTrack } from 'react-native-track-player';
 
 import { Button, PlayerControls, Progress, TrackInfo } from './components';
-import { useCurrentTrack } from './hooks';
 import { QueueInitialTracksService, SetupService } from './services';
 
 const App: React.FC = () => {
-  const track = useCurrentTrack();
+  const track = useActiveTrack();
   const [isPlayerReady, setIsPlayerReady] = useState<boolean>(false);
 
   useEffect(() => {
-    async function run() {
+    let unmounted = false;
```

Comment:
Why do we need this? If the useEffect dependency array is empty then this should only fire once anyway right?

And the unmount callback will destroy the `unmounted` variable state anyway, won't it?

Reply:
For the case where the component was unmounted in the middle of two async calls.